### PR TITLE
Setup notification system to project

### PR DESCRIPTION
**Pull request by ko-yaroslav**

### CODE REVIEW

File: src/translations/en.json [hunk 1]
```diff
@@ -15,5 +15,7 @@
     "errors.invalid.name.length": "From 2 to 30 symbols",
     "auth.forgot.password.link" : "Forgot password",
     "auth.login" : "Log In",
-    "auth.signUp" : "Sign Up"
-}
\ No newline at end of file
+    "auth.signUp" : "Sign Up",
+
+    "notification.title.success": "Success"
```
Comment (yaroslavfedyshyn) [inline src/translations/en.json]: @ko-yaroslav add default translation strings for all of notification types.

Comment (ko-yaroslav) [reply src/translations/en.json]: Done.

File: src/containers/Auth/ForgotPassword/index.js [hunk 2]
```diff
@@ -5,34 +5,46 @@ import { compose } from 'redux';
...
     if (!valid) {
       return touch(...Object.keys(errors));
     }
 
-    const submitter = handleSubmit(async (values) => await onSubmit(values));
+    const submitter = handleSubmit(async (values) => {
+      const forgotPasswordStatus = await onSubmit(values);
+
+      if (forgotPasswordStatus.value && forgotPasswordStatus.value.status === 200) {
+        onSuccessNotification(trans('notification.success.auth.password.forgot'));
```
Comment (yaroslavfedyshyn) [inline src/containers/Auth/ForgotPassword/index.js]: @ko-yaroslav notification should be triggered from story.js. You should not await for end of request in component.

File: src/containers/Auth/Login/index.js [hunk 3]
```diff
@@ -30,7 +31,7 @@ const LoginContainer = (props) => {
     }
 
     const submitter = handleSubmit(async (values) => {
-      await onSubmit(values.toJS());
+      await onSubmit(values.toJS(), props);
```
Comment (yaroslavfedyshyn) [inline src/containers/Auth/Login/index.js]: @ko-yaroslav props shouldn't be passed here to submit fn. If you try to pass sucessNotification fn here = it is wrong. You can just import sucessNotification or any other inside story.js file and dispatch it.

File: src/containers/Auth/Login/index.js [hunk 4]
```diff
@@ -69,7 +70,8 @@ const mapStateToProps = (state) => ({
 });
 
 const mapDispatchToProps = (dispatch) => ({
-  onSubmit: async (data) => await loginStory({ data, dispatch }),
+  onSubmit: async (data, props) => await loginStory({ data, props, dispatch }),
+  onSuccessNotification: (payload) => dispatch(successNotification(payload)),
```
Comment (yaroslavfedyshyn) [inline src/containers/Auth/Login/index.js]: @ko-yaroslav so this should be remove as it is not using in scope of this file

File: src/containers/Auth/ResetPassword/index.js [hunk 5]
```diff
@@ -41,7 +42,7 @@ const ResetPasswordContainer = (props) => {
         ...data
       } = values;
 
-      await onSubmit(data);
+      await onSubmit(data, props);
```
Comment (yaroslavfedyshyn) [inline src/containers/Auth/ResetPassword/index.js]: @ko-yaroslav the same story as on login screen.

File: src/containers/Auth/ResetPassword/index.js [hunk 6]
```diff
@@ -56,14 +57,15 @@ const ResetPasswordContainer = (props) => {
...
+const formName = 'resetPassword';
 
 const mapStateToProps = (state) => ({
   errors: getFormSyncErrors(formName)(state) || {},
 });
 
 const mapDispatchToProps = (dispatch) => ({
-  onSubmit: (payload) => dispatch(resetPasswordAction(payload)),
+  onSubmit: (data, props) => resetPasswordStory({ data, props, dispatch }),
+  onSuccessNotification: (payload) => dispatch(successNotification(payload)),
```
Comment (yaroslavfedyshyn) [inline src/containers/Auth/ResetPassword/index.js]: @ko-yaroslav same. shouldn't be here

File: src/containers/Auth/SignUp/index.js [hunk 7]
```diff
@@ -35,7 +36,7 @@ const SignUpContainer = (props) => {
         ...values
       } = _values.toJS();
 
-      await onSubmit(values);
+      await onSubmit(values, props);
```
Comment (yaroslavfedyshyn) [inline src/containers/Auth/SignUp/index.js]: @ko-yaroslav same as on login screen.

File: src/containers/Auth/SignUp/index.js [hunk 8]
```diff
@@ -65,7 +66,8 @@ const mapStateToProps = (state) => ({
 });
 
 const mapDispatchToProps = (dispatch) => ({
-  onSubmit: async (data) => await signUpStory({ data, dispatch }),
+  onSubmit: async (data, props) => await signUpStory({ data, props, dispatch }),
+  onSuccessNotification: (payload) => dispatch(successNotification(payload)),
```
Comment (yaroslavfedyshyn) [inline src/containers/Auth/SignUp/index.js]: @ko-yaroslav same. remove

File: src/containers/Auth/story.js [hunk 9]
```diff
@@ -3,63 +3,40 @@ import history from '../../history';
...
-export const loginStory = async ({ data, dispatch }) => {
-  await dispatch(loginAction(data));
+export const loginStory = async ({ data, props, dispatch }) => {
+  const trans = (id) => props.intl.formatMessage({ id });
+  const logInStatus = await dispatch(loginAction(data));
 
-  history.push(ROUTE_TO_ROOT);
+  if (logInStatus.value && logInStatus.value.status === 200) {
+    history.push(ROUTE_TO_ROOT);
+    props.onSuccessNotification(trans('notification.success.auth.login'));
```
Comment (yaroslavfedyshyn) [inline src/containers/Auth/story.js]: @ko-yaroslav success notification action should be imported at the beginning of this file and dispatched.

File: src/containers/Auth/story.js [hunk 10]
```diff
@@ -3,63 +3,40 @@ import history from '../../history';
...
 } from './actions';
 import {
   ROUTE_TO_LOGIN,
   ROUTE_TO_ROOT,
 } from '../../constants/routes';
 
 
-export const loginStory = async ({ data, dispatch }) => {
-  await dispatch(loginAction(data));
+export const loginStory = async ({ data, props, dispatch }) => {
```
Comment (yaroslavfedyshyn) [inline src/containers/Auth/story.js]: @ko-yaroslav remove props variable from arguments. There should be only data and dispatch. If you need some additional methods or variables inside story which available only inside component just pass wanted inside data object from component. And then you can parse it like this const { trans, ...requestPayload } = data.

Lets stick to this way of data transfer between story and component.

File: src/containers/Auth/story.js [hunk 11]
```diff
@@ -3,63 +3,40 @@ import history from '../../history';
...
-  history.push(ROUTE_TO_ROOT);
+  if (logInStatus.value && logInStatus.value.status === 200) {
+    history.push(ROUTE_TO_ROOT);
+    props.onSuccessNotification(trans('notification.success.auth.login'));
+  }
 };
 
-export const signUpStory = async ({ data, dispatch }) => {
+export const signUpStory = async ({ data, props, dispatch }) => {
+  const trans = (id) => props.intl.formatMessage({ id });
```
Comment (yaroslavfedyshyn) [inline src/containers/Auth/story.js]: @ko-yaroslav same for props, trans.

File: src/containers/Auth/story.js [hunk 12]
```diff
@@ -3,63 +3,40 @@ import history from '../../history';
...
   const singUpStatus = await dispatch(signUpAction(data));
 
   if (singUpStatus.value && singUpStatus.value.status === 200) {
     history.push(ROUTE_TO_LOGIN);
+    props.onSuccessNotification(trans('notification.success.auth.registration'));
```
Comment (yaroslavfedyshyn) [inline src/containers/Auth/story.js]: @ko-yaroslav import notification action and dispatch

File: src/containers/Auth/story.js [hunk 13]
```diff
@@ -3,63 +3,40 @@ import history from '../../history';
...
-// };
-
-//
-// export const forgotPasswordStory = async ({payload, dispatch}) => {
-//     await dispatch(forgotPasswordAction(payload));
-//
-//     dispatch(successNotification(trans('notification.forgot.password.email.sent')));
-// };
+export const resetPasswordStory = async ({ data, props, dispatch }) => {
+  const trans = (id) => props.intl.formatMessage({ id });
```
Comment (yaroslavfedyshyn) [inline src/containers/Auth/story.js]: @ko-yaroslav same for trans and props

File: src/containers/Auth/story.js [hunk 14]
```diff
@@ -3,63 +3,40 @@ import history from '../../history';
...
-//         setToken(response.data.token);
-//     }
-//
-//     await dispatch(getMeAction());
-//
-//     dispatch(push(redirectTo));
-// };
+  if (resetPasswordStatus.value && resetPasswordStatus.value.status === 200) {
+    history.push(ROUTE_TO_LOGIN);
+    props.onSuccessNotification(trans('notification.success.auth.password.reset'));
```
Comment (yaroslavfedyshyn) [inline src/containers/Auth/story.js]: @ko-yaroslav same for notification

File: src/containers/Auth/story.js [hunk 15]
```diff
@@ -3,63 +3,40 @@ import history from '../../history';
...
-// export const logoutStory = async ({dispatch}) => {
-//     await dispatch(logoutAction());
-//
-//     localStorage.clear();
-//
-//     dispatch(push(ROUTE_TO_LOGIN));
-// };
-
-//
-// export const forgotPasswordStory = async ({payload, dispatch}) => {
```
Comment (yaroslavfedyshyn) [inline src/containers/Auth/story.js]: @ko-yaroslav why did you remove forgotPasswordStory? It should be used as well as other. Also should show notification message as described in task

**Comment by yaroslavfedyshyn** [comment]
@ko-yaroslav go through all of the comment I left for you.

File: src/containers/Auth/Login/index.js (hunk 3 above)
Comment (ko-yaroslav) [reply src/containers/Auth/Login/index.js]: Here I try pass intl

File: src/containers/Auth/ForgotPassword/index.js (hunk 2 above)
Comment (ko-yaroslav) [reply src/containers/Auth/ForgotPassword/index.js]: Done.

File: src/containers/Auth/Login/index.js (hunk 3 above)
Comment (ko-yaroslav) [reply src/containers/Auth/Login/index.js]: Done.

File: src/containers/Auth/Login/index.js (hunk 4 above)
Comment (ko-yaroslav) [reply src/containers/Auth/Login/index.js]: Done.

File: src/containers/Auth/ResetPassword/index.js (hunk 5 above)
Comment (ko-yaroslav) [reply src/containers/Auth/ResetPassword/index.js]: Done.

File: src/containers/Auth/ResetPassword/index.js (hunk 6 above)
Comment (ko-yaroslav) [reply src/containers/Auth/ResetPassword/index.js]: Done.

File: src/containers/Auth/SignUp/index.js (hunk 7 above)
Comment (ko-yaroslav) [reply src/containers/Auth/SignUp/index.js]: Done.

File: src/containers/Auth/SignUp/index.js (hunk 8 above)
Comment (ko-yaroslav) [reply src/containers/Auth/SignUp/index.js]: Done.

File: src/containers/Auth/story.js (hunk 9 above)
Comment (ko-yaroslav) [reply src/containers/Auth/story.js]: Done.

File: src/containers/Auth/story.js (hunk 10 above)
Comment (ko-yaroslav) [reply src/containers/Auth/story.js]: Done.

File: src/containers/Auth/story.js (hunk 11 above)
Comment (ko-yaroslav) [reply src/containers/Auth/story.js]: Done.

File: src/containers/Auth/story.js (hunk 12 above)
Comment (ko-yaroslav) [reply src/containers/Auth/story.js]: Done.

File: src/containers/Auth/story.js (hunk 13 above)
Comment (ko-yaroslav) [reply src/containers/Auth/story.js]: Done.

File: src/containers/Auth/story.js (hunk 14 above)
Comment (ko-yaroslav) [reply src/containers/Auth/story.js]: Done.

File: src/containers/Auth/story.js (hunk 15 above)
Comment (ko-yaroslav) [reply src/containers/Auth/story.js]: Done.
